lib/spack/spack/prompt.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import os
+import textwrap
+
+from llnl.util.tty.color import colorize
+
+from spack.util.environment import EnvironmentModifications
+
+
+def prompt_modifications(prompt, shell, env=os.environ):
```

Comment:
@tgamblin based on your comments I take it this is a bad idea. Do you have any thoughts on how we could make this work?  I could a also just abandon preserving the users prompt for the `--dive`.  I'm okay with that personally. I mainly want a clear indicator you switched to the subshell.